File: Quokka.RISCV.Integration.Tests/CSharp2CPPTranslatorTests/Source/DMA_RegisterTest.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace DMA_RegisterTestSource
{
    public class DMA
    {
        public static DMA Instance { get; set; } = new DMA();

        public virtual int Counter { get; set; }
    }

    public static class Firmware
    {
        public static void EntryPoint()
        {
            while(true)
            {
                DMA.Instance.Counter++;
            }
        }
    }
}

File: Quokka.RISCV.CS2CPP.Tests/Intermediate/FirmwareSource/DMA_RegisterTest.cpp
#include "DMA_RegisterTest.h"
#include <stdint.h>
#include <stdbool.h>
#include "dma.h"
namespace DMA_RegisterTestSource
{
	void Firmware::EntryPoint()
	{
		while(true)
		{
			(DMA_RegisterTestSource_DMA_Counter++);
		}
	}
}
